mongodb/configure.sls

```diff
@@ -0,0 +1,79 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+    - require:
+      - pkg: install_packages
+
+stop_mongodb_service:
+  service.dead:
+    - running: True
+    - name: mongodb
+    - enable: True
+    - require:
+      - pkg: install_packages
+
+place_mongodb_config_file:
+  file.managed:
+    - name: /etc/mongod.conf
+    - template: jinja
+    - source: salt://mongodb/templates/mongod.conf.j2
+    - require:
+      - service: stop_mongodb_service
```

Comment:
As mentioned above, this can probably just be:
```saltstack
    - watch_in:
      - service: mongodb_service_running
```
That will cause mongo to restart whenever this config file changes.